wondrous-app/components/CreateEntity/CreateEntityModal/index.tsx

```diff
@@ -1243,6 +1265,89 @@ export const CreateEntityModal = (props: ICreateEntityModal) => {
 
   const getRoleDataById = (id) => roles?.find((role) => role.id === id);
 
+  const handleSubmitTemplate = (template) => {
```

Comment:
can we move the logic to handle template over to TemplatePicker? I'm thinking the file holds too much logic and it will be difficult to maintain over time but also I think it's better to isolate the logic to the component that consumes it

Reply:
I put it in CreateEntityModal because the form is what autoinputs the values from the template. I cannot manipulate the inputs of the form without having a function that exists in the parent CreateEntityModal